lua/entities/sent_tardis/init.lua

```diff
@@ -1625,6 +1625,418 @@ function ENT:TogglePower()
+			if ent.flightmode then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetMoving"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.moving then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetPilot"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return NULL end
+			if not ent.pilot then return NULL end
+			return ent.pilot
+		end
+		return NULL
+	end,
+
+	["GetHADS"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.hads then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetTracking"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if IsValid(ent.trackingent) then
+				return ent.trackingent
+			else
+				return NULL
+			end
+		end
+		return NULL
+	end,
+
+	["LastAng"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.lastang then
+				return {ent.lastang.p, ent.lastang.y, ent.lastang.r}
+			else
+				return {0,0,0}
+			end
+		end
+		return {0,0,0}
+	end,
+
+	["LastPos"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.lastpos then
+				return ent.lastpos
+			else
+				return Vector()
+			end
+		end
+		return Vector()
+	end,
+
+	["GetSelfrepairing"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.repairing then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetIsomorphic"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.isomorphic then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetPowered"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.power then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetPhyslocked"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.physlocked then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetLocked"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.locked then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end,
+
+	["GetVisible"] = function(ent)
+		if ent and IsValid(ent) then
+			if not (ent:GetClass()=="sent_tardis") then return 0 end
+			if ent.visible then
+				return 1
+			else
+				return 0
+			end
+		end
+		return 0
+	end
+}
+
+function ENT:HandleE2(cmd, ...)
+	if not E2Commands[cmd] then return end
```

Comment:
Should probably hard error here if not valid `error("TARDIS Handler for E2 command " .. cmd .. " is not registered")` or something